api-flask/app/caching.py

```diff
@@ -24,16 +28,33 @@ def cache_file(url, prefix, extension='cache'):
     # If the file exists, return path.
     if os.path.isfile(cache_filepath):
         logger.info('Returning cached file for {}.'.format(url))
-        return cache_filepath
+
+        # if the file is a geotiff, confirm that we can open it.
+        if extension == 'tif':
+            try:
+                rasterio.open(cache_filepath)
+                return cache_filepath
+            except Exception:
+                pass
+        else:
+            return cache_filepath
+
     # If the file does not exist, download and return path.
-    else:
-        r = requests.get(url, verify=False)
+    response = requests.get(url, verify=False)
+
+    try:
```

Comment:
For this case, we could validate the status code we are receiving and then specify the status code we are getting from the remote server. Something like

```
if response.status_code != 200:
  raise InternalServer.. 
```

Reply:
The error status will be specified in the error already. I switched to using the HTTPError exception